ee/localserver/krypto-ec-middleware-response_darwin.go

```diff
@@ -0,0 +1,76 @@
+//go:build darwin
+// +build darwin
+
+package localserver
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log/slog"
+
+	"github.com/kolide/kit/ulid"
+	"github.com/kolide/krypto/pkg/challenge"
+	"github.com/kolide/krypto/pkg/echelper"
+	"github.com/kolide/launcher/ee/secureenclavesigner"
+	"github.com/kolide/launcher/pkg/traces"
+)
+
+func (e *kryptoEcMiddleware) generateChallengeResponse(ctx context.Context, o *challenge.OuterChallenge, data []byte) ([]byte, error) {
+	ctx, span := traces.StartSpan(ctx)
+	defer span.End()
+
+	if e.hardwareSigner == nil || e.hardwareSigner.Public() == nil {
```

Comment:
When/why would the hardware signer be nil? Early on in registration/enrollment before we've generated keys? Or another reason?